phpunit/blocks/render-query-test.php

```diff
@@ -0,0 +1,228 @@
+		<div class="wp-block-query">
+			<!-- wp:post-template {"align":"wide"} -->
+			<!-- /wp:post-template -->
+			<!-- wp:query-pagination -->
+				<!-- wp:query-pagination-previous /-->
+				<!-- wp:query-pagination-next /-->
+			<!-- /wp:query-pagination -->
+		</div>
+		<!-- /wp:query -->
+HTML;
+
+		// Set main query to single post.
+		$wp_query = new WP_Query(
+			array(
+				'posts_per_page' => 1,
+				'paged'          => 2,
+			)
+		);
+
+		$wp_the_query = $wp_query;
+		$prev_paged   = $paged;
+		$paged        = 2;
+
+		$output = do_blocks( $content );
+
+		$paged = $prev_paged;
+
+		$p = new WP_HTML_Tag_Processor( $output );
+
+		$p->next_tag( array( 'class_name' => 'wp-block-query' ) );
+		$this->assertSame( '{"core":{"query":{"loadingText":"Loading page, please wait.","loadedText":"Page Loaded."}}}', $p->get_attribute( 'data-wp-context' ) );
+		$this->assertSame( 'query-0', $p->get_attribute( 'data-wp-navigation-id' ) );
+		$this->assertSame( true, $p->get_attribute( 'data-wp-interactive' ) );
+
+		$p->next_tag( array( 'class_name' => 'wp-block-post' ) );
+		$this->assertSame( 'post-template-item-' . self::$post_2->ID, $p->get_attribute( 'data-wp-key' ) );
+
+		$p->next_tag( array( 'class_name' => 'wp-block-query-pagination-previous' ) );
+		$this->assertSame( 'query-pagination-previous', $p->get_attribute( 'data-wp-key' ) );
+		$this->assertSame( 'actions.core.query.navigate', $p->get_attribute( 'data-wp-on--click' ) );
+		$this->assertSame( 'actions.core.query.prefetch', $p->get_attribute( 'data-wp-on--mouseenter' ) );
+		$this->assertSame( 'effects.core.query.prefetch', $p->get_attribute( 'data-wp-effect' ) );
+
+		$p->next_tag( array( 'class_name' => 'wp-block-query-pagination-next' ) );
+		$this->assertSame( 'query-pagination-next', $p->get_attribute( 'data-wp-key' ) );
+		$this->assertSame( 'actions.core.query.navigate', $p->get_attribute( 'data-wp-on--click' ) );
+		$this->assertSame( 'actions.core.query.prefetch', $p->get_attribute( 'data-wp-on--mouseenter' ) );
+		$this->assertSame( 'effects.core.query.prefetch', $p->get_attribute( 'data-wp-effect' ) );
+
+		$p->next_tag( array( 'class_name' => 'screen-reader-text' ) );
+		$this->assertSame( 'polite', $p->get_attribute( 'aria-live' ) );
+		$this->assertSame( 'context.core.query.message', $p->get_attribute( 'data-wp-text' ) );
+
+		$p->next_tag( array( 'class_name' => 'wp-block-query__enhanced-pagination-animation' ) );
+		$this->assertSame( 'selectors.core.query.startAnimation', $p->get_attribute( 'data-wp-class--start-animation' ) );
+		$this->assertSame( 'selectors.core.query.finishAnimation', $p->get_attribute( 'data-wp-class--finish-animation' ) );
+	}
+
+	/**
+	 * Tests that the `core/query` block adds an extra attribute to disable the
+	 * enhanced pagination in the browser when a plugin block is found inside.
+	 */
+	public function test_rendering_query_with_enhanced_pagination_auto_disabled() {
```

Comment:
If I am correct that post-content should disable EP, then it would be good to duplicate this test to include the content block in place of the plugin block.

Reply:
I answered [below](https://github.com/WordPress/gutenberg/pull/55714#issuecomment-1788635484) and then did it in https://github.com/WordPress/gutenberg/pull/55714/commits/59c0270bba65aa6cff6e0c4e26c851f3e79d051a.